connectors/es/client.py

```diff
@@ -162,6 +165,81 @@ async def wait(self):
         await self.close()
         return False
 
+    async def ping(self):
+        try:
+            await self.client.info()
```

Comment:
Why `ping` is not retried?

Reply:
It's not retried cause of `wait` method - `wait` calls `ping` and does retrying based on its own config options. See [method](https://github.com/elastic/connectors/blob/8e431d529c557e6a6bb3b6a95f2b20b92cb965dd/connectors/es/client.py#L148):

```python
    async def wait(self):
        backoff = self.initial_backoff_duration
        # ...
        while time.time() - start < self.max_wait_duration:
            if not self._keep_waiting:
                await self.close()
                return False

            # ...
            if await self.ping():
                return True
            await self._sleeps.sleep(backoff)
            backoff *= self.backoff_multiplier

        await self.close()
        return False
```